app/models/conversion_host.rb

```diff
@@ -21,31 +26,97 @@ class ConversionHost < ApplicationRecord
 
   include_concern 'Configurations'
 
-  # To be eligible, a conversion host must have the following properties
-  #  - A transport mechanism is configured for source (set by 3rd party)
-  #  - Credentials are set on the resource and SSH connection works
-  #  - The number of concurrent tasks has not reached the limit
+  # Check using all associated authentications if any are directly associated
+  # with the conversion host. Otherwise, use the default check which uses the
+  # associated resource's authentications.
+  #
+  # Subclasses should pass provider-specific +options+, such as proxy information.
+  #
+  # This method is necessary to comply with AuthenticationMixin interface.
+  #--
+  # TODO: Use the verify_credentials_ssh method in host.rb? Move that to the
+  # AuthenticationMixin?
+  #
+  def verify_credentials(_auth_type = nil, options = {})
+    if authentications.empty?
+      check_ssh_connection
+    else
+      require 'net/ssh'
+      host = hostname || ipaddress
+
+      authentications.each do |auth|
```

Comment:
Typically the auth_type is used to pick a single authentication to use rather than looping through all authentications.

If you're looking for username/password you'd use `userid, password = auth_user_pwd(auth_type)`

Reply:
I don't think auth_type would ever be set, though.

Reply:
Alright, I updated it to use the `auth_type`, but it required modifying the authentication mixin because it wasn't looking for private keys as a possible result.